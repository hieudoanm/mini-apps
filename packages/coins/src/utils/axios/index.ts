import axios, { AxiosError, AxiosRequestConfig, AxiosResponse } from 'axios';

export const axiosGet = <T>(
  url: string,
  configs: AxiosRequestConfig = {}
): Promise<T> => {
  return new Promise((resolve, reject) => {
    axios
      .get(url, configs)
      .then((response: AxiosResponse<T>) => {
        return resolve(response.data);
      })
      .catch((error: AxiosError<T>) => {
        return reject(error?.response?.data);
      });
  });
};
