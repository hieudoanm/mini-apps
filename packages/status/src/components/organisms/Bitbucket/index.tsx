import { useAxios } from 'shared';
import { Service, ServiceAPI } from '../../../configs';
import LoadingStatus from '../../molecules/LoadingStatus';

const Bitbucket: React.FC = () => {
  const { loading, error, data, refetch } = useAxios<any>(
    ServiceAPI.BIT_BUCKET
  );

  if (!data) {
    return (
      <LoadingStatus
        loading={loading}
        error={error}
        data={data}
        active={false}
        service={Service.BIT_BUCKET}
        refetch={refetch}
      />
    );
  }

  const active: boolean = data.status.indicator === 'none';

  return (
    <LoadingStatus
      loading={loading}
      error={error}
      data={data}
      active={active}
      service={Service.BIT_BUCKET}
      refetch={refetch}
    />
  );
};

Bitbucket.displayName = 'Bitbucket';

export default Bitbucket;
