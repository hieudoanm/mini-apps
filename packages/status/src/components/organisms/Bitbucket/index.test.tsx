import { render } from '@testing-library/react';
import axios from 'axios';
import Bitbucket from '.';

jest.mock('axios');

describe('Bitbucket', () => {
  test('render default', () => {
    jest.spyOn(axios, 'get').mockResolvedValueOnce({
      data: { data: { status: { indicator: 'none' } } },
    });

    const wrapper = render(<Bitbucket />);
    expect(wrapper.container).toMatchSnapshot();
  });
});
