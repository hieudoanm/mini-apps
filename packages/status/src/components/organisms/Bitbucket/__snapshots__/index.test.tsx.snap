// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bitbucket render default 1`] = `
<div>
  <div
    class="p-4 flex items-center justify-between"
  >
    <div
      class="uppercase"
    >
      Bitbucket
    </div>
    <div
      class="flex items-center gap-2"
    >
      <div
        class="p-1 md:p-2 rounded border rounded-full"
      >
        <div
          class="flex items-center gap-2"
        >
          <p
            class="text-sm hidden md:block leading-4"
          >
            Loading
          </p>
          <div
            class="rounded-full w-3 h-3 md:w-4 md:h-4 bg-gray-500"
          />
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Refresh
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
